component/core/src/main/java/org/exoplatform/social/core/service/LinkProvider.java

```diff
@@ -588,4 +592,61 @@ public static String getResourceBundleLabel(Locale locale, String label) {
     ResourceBundleService resourceBundleService =  ExoContainerContext.getService(ResourceBundleService.class);
     return resourceBundleService.getResourceBundle(resourceBundleService.getSharedResourceBundleNames(), locale).getString(label);
   }
+
+  public static String getBaseURLSiteRest() {
+    return "/" + PortalContainer.getCurrentPortalContainerName() + "/" + PortalContainer.getCurrentRestContextName()
+        + BASE_URL_SITE_REST_API;
+  }
+
+  public static String buildAttachmentUrl(String siteName, String type, boolean isDefault) {
```

Comment:
```suggestion
  public static String buildSiteBannertUrl(PortalConfig site) {
```
try to call the private existing method buildAttachmentUrl with the needed site params